tests/wakunode_rest/test_rest_admin.nim

```diff
@@ -133,8 +124,16 @@ suite "Waku v2 Rest API - Admin":
       getRes.status == 200
       $getRes.contentType == $MIMETYPE_JSON
       getRes.data.len() == 2
-      ($getRes.data).contains(expectedFilterData2)
-      ($getRes.data).contains(expectedFilterData3)
+
+    let
+      peers = @[getRes.data[0].peerId, getRes.data[1].peerId]
+      numCriteria = @[getRes.data[0].filterCriteria.len, getRes.data[1].filterCriteria.len]
+
+    check:
+      $peerInfo2 in peers
+      $peerInfo3 in peers
+      2 in numCriteria
+      3 in numCriteria
```

Comment:
Is it enough to only check the number of criteria? I guess the filter tests have this covered right?

Reply:
If the data array is of size 2, then numCriteria (as defined now) will always have size 2, there's no need to check that.

So what we check here is that of the 2 peers that we have, one has 2 subscriptions and the other 3.

I agree that Filter tests kind of cover this scenario, but not completely - because here we're testing the output from the REST API handler, and not the internal data structures. The REST API can return wrong values even if Filter itself works fine 😵‍💫